jwst/assign_wcs/nircam.py

```diff
@@ -190,13 +190,14 @@ def tsgrism(input_model, reference_files):
     # input into the forward transform is x,y,x0,y0,order
     # where x,y is the pixel location in the grism image
     # and x0,y0 is the source location in the "direct" image
-    # For this mode, the source is always at crpix1,crpis2
-    # discussion with nadia that wcsinfo might not be available
+    # For this mode, the source is always at crpix1 x crpix2, which
+    # are stored in keywords XREF_SCI, YREF_SCI.
+    # Discussion with nadia that wcsinfo might not be available
     # here but crpix info could be in wcs.source_location or similar
     # TSGRISM mode places the sources at crpix, and all subarrays
     # begin at 0,0, so no need to translate the crpix to full frame
     # because they already are in full frame coordinates.
-    xc, yc = (input_model.meta.wcsinfo.crpix1, input_model.meta.wcsinfo.crpix2)
+    xc, yc = (input_model.meta.wcsinfo.siaf_xref_sci, input_model.meta.wcsinfo.siaf_yref_sci)
```

Comment:
Will this cause any old regression tests to fail?

Reply:
It resulted in one unit test that needed to be updated (to also switch from crpixn to x/yref_sci), but not sure about regression tests. According to JP-811, we don't have any existing tests for NIRCam TS-GRISM mode (just NIRCam TS-Image). So I'm guessing we won't see any failures.